x-pack/legacy/plugins/siem/public/components/ml_popover/types.ts

```diff
@@ -12,26 +12,103 @@ export interface Group {
   calendarIds: string[];
 }
 
+export interface CheckRecognizerProps {
+  indexPatternName: string;
+  headers?: Record<string, string | undefined>;
+  signal: AbortSignal;
+}
+
+export interface RecognizerModule {
+  id: string;
+  title: string;
+  query: Record<string, object>;
+  description: string;
+  logo: {
+    icon: string;
+  };
+}
+
+export interface GetModulesProps {
+  moduleId?: string;
+  headers?: Record<string, string | undefined>;
+  signal: AbortSignal;
+}
+
+export interface Module {
+  id: string;
+  title: string;
+  description: string;
+  type: string;
+  logoFile: string;
+  defaultIndexPattern: string;
+  query: Record<string, object>;
+  jobs: ModuleJob[];
+  datafeeds: ModuleDatafeed[];
+  kibana: object;
+}
+
+/**
+ * Representation of an ML Job as returned from `the ml/modules/get_module` API
+ */
+export interface ModuleJob {
+  id: string;
+  config: {
+    groups: string[];
+    description: string;
+    analysis_config: {
+      bucket_span: string;
+      summary_count_field_name?: string;
+      detectors: Detector[];
+      influencers: string[];
+    };
+    analysis_limits: {
+      model_memory_limit: string;
+    };
+    data_description: {
+      time_field: string;
+      time_format?: string;
+    };
+    model_plot_config?: {
+      enabled: boolean;
+    };
+    custom_settings: {
+      created_by: string;
+      custom_urls: CustomURL[];
+    };
+    job_type: string;
+  };
+}
+
+export interface ModuleDatafeed {
+  id: string;
+  config: {
+    job_id: string;
+    // TODO: Speak to ML team about this one
```

Comment:
I would add that sometimes the API returns `indexes` and sometimes it returns `indices` as the reason for the communications needed in case it takes a while before this is fixed.